[Refactor] Disable column pool in shared_data mode

stonechen111 opened the pull request:
In order to avoid behavior changes, the modification of disable column pool is placed in be heartbeat.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [x] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

kevincai commented on be/src/agent/heartbeat_server.cpp:
```diff
@@ -131,6 +131,12 @@ void HeartbeatServer::heartbeat(THeartbeatResult& heartbeat_result, const TMaste
+    // Temporary solution for modifying the default value of disable_column_pool in lower versions
+    // to avoid behavior changes.
+    if (!config::disable_column_pool && master_info.run_mode == TRunMode::SHARED_DATA) {
+        config::disable_column_pool = true;
```
at least add one line log, indicate that the configuration is turned on explicitly in shared data mode.

stonechen111 replied:
ok

stonechen111 committed:
[Refactor] Disable column pool in shared_data mode

In order to avoid behavior changes, the modification of disable column
pool is placed in be heartbeat.

Signed-off-by: stonechen111 <chenshi@starrocks.com>
